pkg/deploy/submariner.go

```diff
@@ -0,0 +1,213 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deploy
+
+import (
+	"encoding/base64"
+	submariner "github.com/submariner-io/submariner-operator/api/submariner/v1alpha1"
+	"github.com/submariner-io/submariner-operator/internal/constants"
+	"github.com/submariner-io/submariner-operator/pkg/broker"
+	"github.com/submariner-io/submariner-operator/pkg/client"
+	"github.com/submariner-io/submariner-operator/pkg/discovery/globalnet"
+	"github.com/submariner-io/submariner-operator/pkg/reporter"
+	"github.com/submariner-io/submariner-operator/pkg/secret"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/operator/servicediscoverycr"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/operator/submarinercr"
+	v1 "k8s.io/api/core/v1"
+	"strings"
+)
+
+func Submariner(clientProducer client.Producer, jo WithJoinOptions, brokerInfo broker.Info, brokerSecret *v1.Secret, netconfig globalnet.Config, imageOverrides map[string]string, status reporter.Interface) error {
```

Comment:
I think we should define a specific `SubmarinerOptions` instead of using the broader `WithJoinOptions`.

Reply:
THere are few options common to both submariner and servicediscovery

Reply:
Yes but  most aren't and some don't apply to either `Submariner` or `ServiceDiscovery`. I think it would be cleaner to define separate `Options` types for each and to avoid unnecessary coupling.